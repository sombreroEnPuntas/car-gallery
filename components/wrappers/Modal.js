import styled from 'styled-components'

const Modal = styled.div`
  align-items: center;
  background-color: rgba(0, 0, 0, 0.4);
  display: ${({ open }) => (open ? 'flex' : 'none')};
  height: 100%;
  justify-content: center;
  left: 0;
  overflow: auto;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
`

export default Modal
