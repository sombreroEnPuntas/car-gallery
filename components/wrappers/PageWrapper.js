import styled from 'styled-components'
import { breakpoints } from '.'

const PageWrapper = styled.div`
  margin: auto;
  width: 96vw;

  @media (min-width: ${breakpoints['mobile']}) {
    width: 80vw;
  }
`

export default PageWrapper
