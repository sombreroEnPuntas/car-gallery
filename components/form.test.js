// @flow
import React from 'react'
import { shallow } from 'enzyme'

// Dependencies
import PageWrapper from './wrappers/PageWrapper'
import Modal from './wrappers/Modal'

// Tested Unit
import TestedComponent from './form'

const getProps = customProps => ({
  children: <div>{':)'}</div>,
  isLoading: false,
  message: null,
  ...customProps,
})

describe('Message', () => {
  const wrapper = shallow(<TestedComponent {...getProps()} />)

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('shows children', () => {
    const actual = wrapper.find(PageWrapper).text()
    const expected = ':)'

    expect(actual).toEqual(expected)
  })

  it('opens Modal when loading', () => {
    wrapper.setProps({ isLoading: true, message: null })

    expect(
      wrapper
        .find(Modal)
        .dive()
        .prop('open')
    ).toBe(true)
  })

  it('opens Modal when error', () => {
    wrapper.setProps({ isLoading: false, message: `418 I'm a teapot` })

    expect(
      wrapper
        .find(Modal)
        .dive()
        .prop('open')
    ).toBe(true)
  })
})
