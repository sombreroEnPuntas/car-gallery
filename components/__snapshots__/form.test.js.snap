// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message matches the snapshot 1`] = `
<Fragment>
  <PageWrapper>
    <div>
      :)
    </div>
  </PageWrapper>
  <Modal
    open={false}
  >
    <Message
      message={null}
    />
  </Modal>
</Fragment>
`;
